src/peft/tuners/lora/eva.py

```diff
@@ -571,29 +591,4 @@ def prepare_layer_inputs_fn_language_modeling(layer_input, model_input, layer_na
             show_progress_bar=show_progress_bar,
         )
 
-    # assert all lora layers are contained in eva_state_dict
-    update_layer_kwargs = {
-        "adapter_name": adapter_name,
-        "lora_alpha": peft_config.lora_alpha,
-        "lora_dropout": peft_config.lora_dropout,
-        "use_rslora": peft_config.use_rslora,
-        "use_dora": peft_config.use_dora,
-    }
-    missing_eva_inits = []
-    for name, module in model.named_modules():
-        if not isinstance(module, LoraLayer):
-            continue
-        if name in eva_state_dict:
-            w = eva_state_dict.pop(name)
-            module.update_layer(r=w.size(0), init_lora_weights=peft_config.init_lora_weights, **update_layer_kwargs)
-            module.lora_A[adapter_name].weight.copy_(w)
-        else:
-            module.update_layer(r=peft_config.r, init_lora_weights=True, **update_layer_kwargs)
-            missing_eva_inits.append(name)
-
-    if missing_eva_inits:
-        warnings.warn(
-            f"the following layers were initialized with init_lora_weights=True because they were not found in the eva state_dict: {missing_eva_inits} "
-            f"currently the following lora modules are not supported: {UNSUPPORTED_LORA_MODULES}"
-        )
-    return model
+    load_eva_state_dict(model, eva_state_dict, adapter_name, _check_valid=False)
```

Comment:
I did it this way because when call `init_lora_eva_weights` we would otherwise check validity twice, once inside `init_lora_eva_weights` and then again inside `load_eva_state_dict` which is called inside `init_lora_eva_weights`. For that reason I defined the private argument but I am also not super keen on keeping it this way. I could remove it and just accept that the checks are run twice. Or maybe there is a better way to do it.